mod_modular_vanilla/hooks/entity/tactical/player.nut

```diff
@@ -67,8 +67,9 @@
 				if (potential.len() == 0)
 					break;
 
-				trait = ::new(::MSU.Array.rand(potential)[1]);
-				traits.push(trait);
+				local randomPick = ::MSU.Array.rand(potential);
+				::MSU.Array.remove(potential, randomPick);
+				traits.push(::new(randomPick[1]));
```

Comment:
Two problems:
1. It doesn't set the `trait` variable to the new trait so line 65 will not work properly.
2. It is a bit inefficient because of first rolling and then iterating over the array to remove the value. I'd do the following:
```suggestion
				trait = ::new(potential.remove(::Math.rand(0, potential.len() - 1))[1]);
				traits.push(trait);
```
or to make it a bit more readable:
```suggestion
				local choice = potential.remove(::Math.rand(0, potential.len() - 1));
				trait = ::new(choice[1]);
				traits.push(trait);
```
Choose whichever option you like more.